src/Processors/Transforms/DistinctSortedChunkTransform.h

```diff
@@ -32,23 +32,27 @@ class DistinctSortedChunkTransform : public ISimpleTransform
         const SizeLimits & output_size_limits_,
         UInt64 limit_hint_,
         const SortDescription & sorted_columns_descr_,
-        const Names & source_columns_);
+        const Names & source_columns_,
+        bool sorted_stream_);
 
-    String getName() const override { return "DistinctSortedChunkTransform"; }
+    String getName() const override { return (!sorted_stream ? "DistinctSortedChunkTransform" : "DistinctSortedStreamTransform"); }
```

Comment:
Looks like `sorted_stream` field` affects only `getName` behaviour. Do we really need it?

Reply:
Yes, it affects only the name. `DistinctSortedStream` just reflects better what is doing when instantiated for final distinct